docs/databases/hadoop/install-and-configure-hadoop-cluster.md

```diff
@@ -0,0 +1,519 @@
+
+    This is configured in `mapred-site.xml` with property `yarn.app.mapreduce.am.resource.mb`.
+
+4. How much memory each map or reduce operation will be allocated. Again, this should be less than the container maximum size
+
+    This is configured in `mapred-site.xml` with properties `mapreduce.map.memory.mb` and `mapreduce.reduce.memory.mb`
+
+The relations between all those properties can be seen in the following figure:
+
+![Schema of memory allocation properties](/docs/assets/hadoop/hadoop-2-memory-allocation-wide.png)
+
+### Sample Configuration for 2Gb nodes
+
+For 2Gb nodes, a working configuration may be:
+
+{: .table .table-striped}
+| Left-Aligned | Centered |
+| ---------------- |:-------------:|
+| yarn.nodemanager.resource.memory-mb     | 1536          |
+| yarn.scheduler.maximum-allocation-mb| 1536 |
+| yarn.scheduler.minimum-allocation-mb| 128 |
+| yarn.app.mapreduce.am.resource.mb| 512 |
+| mapreduce.map.memory.mb| 256 |
+| mapreduce.reduce.memory.mb| 256 |
+
+
+1. Edit `/home/hadoop/hadoop/etc/hadoop/yarn-site.xml` and add the following lines:
+
+    {: .file }
+    ~/hadoop/etc/hadoop/yarn-site.xml
+    :   ~~~ xml
+        <property>
+                <name>yarn.nodemanager.resource.memory-mb</name>
+                <value>1536</value>
+        </property>
+
+        <property>
+                <name>yarn.scheduler.maximum-allocation-mb</name>
+                <value>1536</value>
+        </property>
+
+        <property>
+                <name>yarn.scheduler.minimum-allocation-mb</name>
+                <value>128</value>
+        </property>
+
+        <property>
+                <name>yarn.nodemanager.vmem-check-enabled</name>
+                <value>false</value>
+        </property>
+        ~~~
+
+    The last property disable virtual memory checking that can prevent containers to be allocated properly on JDK8.
+
+
+2. Edit `/home/hadoop/hadoop/etc/hadoop/mapred-site.xml` and add the following lines:
+
+    {: .file }
+    ~/hadoop/etc/hadoop/mapred-site.xml
+    :   ~~~ xml
+        <property>
+                <name>yarn.app.mapreduce.am.resource.mb</name>
+                <value>512</value>
+        </property>
+
+        <property>
+                <name>mapreduce.map.memory.mb</name>
+                <value>256</value>
+        </property>
+
+        <property>
+                <name>mapreduce.reduce.memory.mb</name>
+                <value>256</value>
+        </property>
+        ~~~
+
+
+## Duplicate configuration files to each nodes
+
+1. Copy hadoop binaries to slave nodes
+
+        cd /home/hadoop/
+        scp hadoop-*.tar.gz node1:/home/hadoop
+        scp hadoop-*.tar.gz node2:/home/hadoop
+
+2. Connect to node1 via ssh (no password required thanks to the ssh keys copyied above):
+
+        ssh node1
+
+3. Unzip the binaries, rename the directory, and exit node1 to get back on node-master:
+
+        tar -xzf hadoop-2.8.1.tar.gz
+        ln -s hadoop-2.8.1 hadoop
```

Comment:
mv hadoop-2.8.1 hadoop

(symbolic link only add complexity, they are not needed)